projects/plugins/jetpack/modules/theme-tools/site-logo.php

```diff
@@ -44,3 +44,25 @@ function site_logo_init() {
 	}
 }
 add_action( 'init', 'site_logo_init' );
+
+/**
+ * Transforms the legacy site_logo array, when present, into an attachment ID.
+ *
+ * The attachment ID is the format used for the site_logo option by the Site Logo block,
+ * and the updated Jetpack site-logo feature.
+ *
+ * @since 9.9.0
+ *
+ * @param int|array $site_logo Option.
+ * @return int
+ */
+function jetpack_site_logo_block_compat( $site_logo ) {
+	if ( isset( $site_logo['id'] ) ) {
+		remove_filter( 'option_site_logo', 'jetpack_site_logo_block_compat', 1 );
+		update_option( 'site_logo', $site_logo['id'] );
+		return $site_logo['id'];
+	}
+
+	return $site_logo;
+}
+add_filter( 'option_site_logo', 'jetpack_site_logo_block_compat', 1 );
```

Comment:
This transforms the `site_logo` option from an associative array with `id`, `sizes`, and `url`, to only an attachment ID. The sizes aren't actually used anymore, and the `url` can be derived from the attachment ID with `wp_get_attachment_url`, so we're not actually losing any data with this change.

Reply:
Would it be worth making the priority a little higher than 1 so someone can easily register something before it in future?

Reply:
The intention is to run this early as possible, before anything else accesses it to make sure it's updated before being used.

If someone wanted to precede it, I believe they could use a priority of `0` or a negative priority and that would run before this.